compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -1614,4 +1615,20 @@ object messages {
     val msg = s"super call not allowed in inline $symbol"
     val explanation = "Method inlining prohibits calling superclass methods, as it may lead to confusion about which super is being called."
   }
+
+  case class ModifiersNotAllowed(flags: FlagSet, sort: String)(implicit ctx: Context)
+    extends Message(ModifiersNotAllowedID) {
+    val kind = "Syntax"
+    val msg = s"modifier(s) `$flags' not allowed for $sort"
+    val explanation = {
+      val code = "sealed def y: Int = 1"
+      hl"""You tried to use a modifier that is inapplicable for the type of item under modification
+         |
+         |
+         |Consider the following example:
+         |$code
+         |In this instance, the modifier 'sealed' is not applicable to the item type 'def' (method)
```

Comment:
Hmm, I think these two should be harmonized. I.e. the same error message for both. You could make `sort: Option[String]` to discern between the `case _` and the rest of them.

Otherwise this LGTM now :)